From 24fa3e59c670dfbe8f2e4dbfc68b1a5e22f633f6 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Tue, 6 Aug 2013 12:47:45 -0700
Subject: [PATCH 207/974] MIPS: OCTEON: Use current_cpu_type() for CPU model
 check.

Replace OCTEON_IS_OCTEON2() and OCTEON_IS_OCTEON3() with appropriate
current_cpu_type().

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/cavium-octeon/dma-octeon.c            | 5 +++--
 arch/mips/cavium-octeon/octeon-platform.c       | 2 +-
 arch/mips/cavium-octeon/octeon-power-throttle.c | 3 ++-
 arch/mips/cavium-octeon/perf_counters.c         | 2 +-
 arch/mips/cavium-octeon/setup.c                 | 6 +++---
 drivers/edac/octeon_edac-lmc.c                  | 5 +++--
 drivers/edac/octeon_edac-pc.c                   | 2 +-
 7 files changed, 14 insertions(+), 11 deletions(-)

diff --git a/arch/mips/cavium-octeon/dma-octeon.c b/arch/mips/cavium-octeon/dma-octeon.c
index 51673c4..bad65db 100644
--- a/arch/mips/cavium-octeon/dma-octeon.c
+++ b/arch/mips/cavium-octeon/dma-octeon.c
@@ -276,7 +276,8 @@ void __init plat_swiotlb_setup(void)
 			continue;
 
 		/* These addresses map low for PCI. */
-		if (e->addr > 0x410000000ull && !OCTEON_IS_OCTEON2())
+		if (e->addr > 0x410000000ull &&
+		    current_cpu_type() != CPU_CAVIUM_OCTEON2)
 			continue;
 
 		addr_size += e->size;
@@ -308,7 +309,7 @@ void __init plat_swiotlb_setup(void)
 #endif
 #ifdef CONFIG_USB_OCTEON_OHCI
 	/* OCTEON II ohci is only 32-bit. */
-	if (OCTEON_IS_OCTEON2() && max_addr >= 0x100000000ul)
+	if (current_cpu_type() == CPU_CAVIUM_OCTEON2 && max_addr >= 0x100000000ul)
 		swiotlbsize = 64 * (1<<20);
 #endif
 	swiotlb_nslabs = swiotlbsize >> IO_TLB_SHIFT;
diff --git a/arch/mips/cavium-octeon/octeon-platform.c b/arch/mips/cavium-octeon/octeon-platform.c
index 1f9a91a..1fd6ad6 100644
--- a/arch/mips/cavium-octeon/octeon-platform.c
+++ b/arch/mips/cavium-octeon/octeon-platform.c
@@ -593,7 +593,7 @@ end_led:
 	if (alias_prop) {
 		int usbn = fdt_path_offset(initial_boot_params, alias_prop);
 
-		if (usbn >= 0 && (OCTEON_IS_OCTEON2() ||
+		if (usbn >= 0 && (current_cpu_type() == CPU_CAVIUM_OCTEON2 ||
 				  !octeon_has_feature(OCTEON_FEATURE_USB))) {
 			pr_debug("Deleting usbn\n");
 			fdt_nop_node(initial_boot_params, usbn);
diff --git a/arch/mips/cavium-octeon/octeon-power-throttle.c b/arch/mips/cavium-octeon/octeon-power-throttle.c
index ba60303..0759c81 100644
--- a/arch/mips/cavium-octeon/octeon-power-throttle.c
+++ b/arch/mips/cavium-octeon/octeon-power-throttle.c
@@ -198,7 +198,8 @@ static __init int octeon_power_throttle_init(void)
 	unsigned int cpu = 0;
 	int err = 0;
 
-	if (!(OCTEON_IS_OCTEON2() || OCTEON_IS_OCTEON3()))
+	if (!(current_cpu_type() == CPU_CAVIUM_OCTEON2 ||
+		current_cpu_type() == CPU_CAVIUM_OCTEON3))
 		return 0;
 
 	get_online_cpus();
diff --git a/arch/mips/cavium-octeon/perf_counters.c b/arch/mips/cavium-octeon/perf_counters.c
index 60718b8..171b497 100644
--- a/arch/mips/cavium-octeon/perf_counters.c
+++ b/arch/mips/cavium-octeon/perf_counters.c
@@ -384,7 +384,7 @@ static int proc_perf_show(struct seq_file *m, void *v)
 
 			/* Check if LMC controller is enabled. */
 			ctl2.u64 = cvmx_read_csr(CVMX_LMCX_DLL_CTL2(tad));
-			if (OCTEON_IS_OCTEON3()) {
+			if (current_cpu_type() == CPU_CAVIUM_OCTEON3) {
 				if (ctl2.cn70xx.quad_dll_ena == 0)
 					continue;
 			} else if (ctl2.cn63xx.quad_dll_ena == 0)
diff --git a/arch/mips/cavium-octeon/setup.c b/arch/mips/cavium-octeon/setup.c
index ad888bc..130fbb9 100644
--- a/arch/mips/cavium-octeon/setup.c
+++ b/arch/mips/cavium-octeon/setup.c
@@ -502,7 +502,7 @@ void octeon_user_io_init(void)
 	cvmmemctl.s.cvmsegenau = 0;
 
 	/* Enable TLB parity error reporting on OCTEON II */
-	if (OCTEON_IS_OCTEON2())
+	if (current_cpu_type() == CPU_CAVIUM_OCTEON2)
 		cvmmemctl.s.tlbperrena = 1;
 
 	write_c0_cvmmemctl(cvmmemctl.u64);
@@ -741,12 +741,12 @@ void __init prom_init(void)
 	sysinfo->dfa_ref_clock_hz = octeon_bootinfo->dfa_ref_clock_hz;
 	sysinfo->bootloader_config_flags = octeon_bootinfo->config_flags;
 
-	if (OCTEON_IS_OCTEON2()) {
+	if (current_cpu_type() == CPU_CAVIUM_OCTEON2) {
 		/* I/O clock runs at a different rate than the CPU. */
 		union cvmx_mio_rst_boot rst_boot;
 		rst_boot.u64 = cvmx_read_csr(CVMX_MIO_RST_BOOT);
 		octeon_io_clock_rate = 50000000 * rst_boot.s.pnr_mul;
-	} else if (OCTEON_IS_OCTEON3()) {
+	} else if (current_cpu_type() == CPU_CAVIUM_OCTEON3) {
 		/* I/O clock runs at a different rate than the CPU. */
 		union cvmx_rst_boot rst_boot;
 		rst_boot.u64 = cvmx_read_csr(CVMX_RST_BOOT);
diff --git a/drivers/edac/octeon_edac-lmc.c b/drivers/edac/octeon_edac-lmc.c
index 12d1510..fade62a 100644
--- a/drivers/edac/octeon_edac-lmc.c
+++ b/drivers/edac/octeon_edac-lmc.c
@@ -134,8 +134,9 @@ static int octeon_lmc_edac_probe(struct platform_device *pdev)
 
 		/* Check if LMC controller is enabled. */
 		ctl2.u64 = cvmx_read_csr(CVMX_LMCX_DLL_CTL2(mc));
-		if ((OCTEON_IS_OCTEON3() && ctl2.cn70xx.quad_dll_ena == 0)
-		    || (OCTEON_IS_OCTEON2() && ctl2.cn63xx.quad_dll_ena == 0)) { 
+		if ((current_cpu_type() == CPU_CAVIUM_OCTEON3 && ctl2.cn70xx.quad_dll_ena == 0)
+		    || (current_cpu_type() == CPU_CAVIUM_OCTEON2
+			&& ctl2.cn63xx.quad_dll_ena == 0)) { 
 			dev_info(&pdev->dev, "Disabled (LMC not present)\n");
 			return 0;
 		}
diff --git a/drivers/edac/octeon_edac-pc.c b/drivers/edac/octeon_edac-pc.c
index 65bb001..4f0ac09 100644
--- a/drivers/edac/octeon_edac-pc.c
+++ b/drivers/edac/octeon_edac-pc.c
@@ -86,7 +86,7 @@ static int  co_cache_error_event(struct notifier_block *this,
 			edac_device_handle_ce(p->ed, cpu, 0, "dcache");
 
 		/* Clear the error indication */
-		if (OCTEON_IS_OCTEON2())
+		if (current_cpu_type() == CPU_CAVIUM_OCTEON2)
 			write_octeon_c0_dcacheerr(1);
 		else
 			write_octeon_c0_dcacheerr(0);
-- 
2.6.2

